packages/php-wasm/web/src/lib/web-php.ts

```diff
@@ -44,6 +48,560 @@ const fakeWebsocket = () => {
 	};
 };
 
+async function generateKeys() {
+	const keyPair = await crypto.subtle.generateKey(
+		{
+			name: 'ECDH',
+			namedCurve: 'P-256',
+		},
+		true,
+		['deriveKey']
+	);
+	return keyPair;
+}
+
+function createCertificate({ commonName }) {
+	// create certificate
+	var pki = forge.pki;
+	var keys = pki.rsa.generateKeyPair(2048);
+	var cert = pki.createCertificate();
+	cert.publicKey = keys.publicKey;
+	// alternatively set public key from a csr
+	//cert.publicKey = csr.publicKey;
+	// NOTE: serialNumber is the hex encoded value of an ASN.1 INTEGER.
+	// Conforming CAs should ensure serialNumber is:
+	// - no more than 20 octets
+	// - non-negative (prefix a '00' if your value starts with a '1' bit)
+	cert.serialNumber = '01';
+	cert.validity.notBefore = new Date();
+	cert.validity.notAfter = new Date();
+	cert.validity.notAfter.setFullYear(
+		cert.validity.notBefore.getFullYear() + 1
+	);
+	var attrs = [
+		{
+			name: 'commonName',
+			value: commonName,
+		},
+		{
+			name: 'countryName',
+			value: 'US',
+		},
+		{
+			shortName: 'ST',
+			value: 'Virginia',
+		},
+		{
+			name: 'localityName',
+			value: 'Blacksburg',
+		},
+		{
+			name: 'organizationName',
+			value: 'Test',
+		},
+		{
+			shortName: 'OU',
+			value: 'Test',
+		},
+	];
+	cert.setSubject(attrs);
+	// alternatively set subject from a csr
+	//cert.setSubject(csr.subject.attributes);
+	cert.setIssuer(attrs);
+	cert.setExtensions([
+		{
+			name: 'basicConstraints',
+			cA: true,
+		},
+		{
+			name: 'keyUsage',
+			keyCertSign: true,
+			digitalSignature: true,
+			nonRepudiation: true,
+			keyEncipherment: true,
+			dataEncipherment: true,
+		},
+		{
+			name: 'extKeyUsage',
+			serverAuth: true,
+			clientAuth: true,
+			codeSigning: true,
+			emailProtection: true,
+			timeStamping: true,
+		},
+		{
+			name: 'nsCertType',
+			client: true,
+			server: true,
+			email: true,
+			objsign: true,
+			sslCA: true,
+			emailCA: true,
+			objCA: true,
+		},
```

Comment:
we surely don't need all these capabilities in the certificate do we? it shouldn't hurt anything, but do we expect, for example, the Playground to sign email signatures?

Reply:
Good point, most of these should be fine to remove